### ISSUE TITLE
Doom Emacs code suggestions do not work for std, works for rls

### ISSUE THREAD

**Issue by Schievel1** [rust-analyzer/rust-analyzer#11580]
**rust-analyzer version**: (eg. output of "Rust Analyzer: Show RA Version" command)

**rustc version**: (eg. output of `rustc -V`)

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTUP_HOME` or `CARGO_HOME`)

**Comment by flodiebold** [comment]
Is there any output in the `*rust-analyzer: stderr*` buffer? What does `ls $(rustc --print sysroot)/lib/rustlib/src/rust/library` print?

**Comment by apmanol** [comment]
Similar issue here with the upgrade to the latest:
```
rust-analyzer --version 
rust-analyzer 4e72700e3 2022-02-28 dev
```
and plain `emacs` with [lsp](https://emacs-lsp.github.io/lsp-mode/page/lsp-rust-analyzer/) and `rustic-mode` 
```
cargo init tttt
cd ttt && emacs src/main.rs 
LSP :: Connected to [rust-analyzer:28488/starting]. [2 times]
LSP :: rust-analyzer:28488 initialized successfully in folders: (/tmp/ttt)
LSP :: Error from the Language Server: waiting for cargo metadata or cargo check (Unknown error) [8 times]
```

There is nothing `*rust-analyzer: stderr*` 

```
ls $(rustc --print sysroot)/lib/rustlib/src/rust/library 
alloc  backtrace  core  panic_abort  panic_unwind  portable-simd  proc_macro  profiler_builtins  rtstartup  rustc-std-workspace-alloc  rustc-std-workspace-core  rustc-std-workspace-std  std  stdarch  test  unwind
```

**Comment by flodiebold** [comment]
Do you get completions for e.g. local variables? You only mentioned snippets.

**Comment by TheEros** [comment]
Normal output does not affect use

**Comment by Schievel1** [comment]
> Is there any output in the `*rust-analyzer: stderr*` buffer? What does `ls $(rustc --print sysroot)/lib/rustlib/src/rust/library` print?

Nothing in *rust-analyzer: stderr*. 

`ls $(rustc --print sysroot)/lib/rustlib/src/rust/library`
```
alloc      core         panic_unwind   proc_macro         rtstartup                  rustc-std-workspace-core  std      test
backtrace  panic_abort  portable-simd  profiler_builtins  rustc-std-workspace-alloc  rustc-std-workspace-std   stdarch  unwind
```

**Comment by Schievel1** [comment]
> Do you get completions for e.g. local variables? You only mentioned snippets.

no, only keywords and snippets

**Comment by Schievel1** [comment]
I get a completion for if:
![Screenshot from 2022-02-28 12-08-16](https://user-images.githubusercontent.com/85408625/155973290-5f3aef96-ad0c-4f07-a067-ce735b1d0a90.png)
but nothing for a local variable or for println!:
![Screenshot from 2022-02-28 12-08-54](https://user-images.githubusercontent.com/85408625/155973397-94bafa69-3f5c-4851-9a32-5122026009c5.png)

**Comment by Schievel1** [comment]
also here is the result from `lsp-toggle-trace-io`, typing prin, deleting it then typing if and enter.
https://pastebin.com/x9zzS3ai

**Comment by bjorn3** [comment]
> LSP :: Error from the Language Server: waiting for cargo metadata or cargo check (Unknown error) [8 times]

What does running `cargo metadata` show?

**Comment by Schievel1** [comment]
> > LSP :: Error from the Language Server: waiting for cargo metadata or cargo check (Unknown error) [8 times]
> 
> What does running `cargo metadata` show?
https://pastebin.com/pYykNhrX

**Comment by Schievel1** [comment]
I also installed toolchain nightly, installed rust-src and clippy for it, `unset RUST_SRC_PATH`, did `export PATH="/home/pascal/.cargo/bin:$PATH" ` and `doom env`.
Same result. 
Same when it set  export `RUST_SRC_PATH=/home/pascal/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library` and `doom env` again.

**Comment by Schievel1** [comment]
wait a sec. this has something to do with rustlings. If I do `cargo new ./bla` and edit ./bla/src/main.rs it suggests println! just fine.

**Comment by lnicola** [comment]
Yeah, Rustlings has a weird project structure, it doesn't work very well with RA (and had trouble with RLS in the past, I think).

**Comment by Schievel1** [comment]
sorry about that then. And thanks for your time and help. 
 I am going to close this here.

**Comment by jackos** [comment]
You can do this to fix your project to work with rust analyzer until that PR is merged:

```bash
# Install rustlings-fix from cargo
cargo install rustlings-fix

# Change directory into wherever rustlings is cloned
cd ~/src/rustlings

# Run the binary
rustlings-fix
```

**Comment by Schievel1** [comment]
> You can do this to fix your project to work with rust analyzer until that PR is merged:
> 
> ```shell
> # Install rustlings-fix from cargo
> cargo install rustlings-fix
> 
> # Change directory into wherever rustlings is cloned
> cd ~/src/rustlings
> 
> # Run the binary
> rustlings-fix
> ```

works like a charm. you guys are awesome! thanks

but now I have to learn some rust then instead of  playing around with emacs all day long... :D
